torch/distributed/pipelining/PipelineStage.py

```diff
@@ -581,7 +581,9 @@ def _validate_fwd_input(self, args, kwargs):
             # TODO why is there a separate recv_info for each pipeline chunk?
             expected_args = self.args_recv_info[self.fwd_chunk_id]
         else:
-            expected_args = tuple()
+            # We don't check inputs for non-0 stages assuming they don't accept
```

Comment:
nit: we actually don't support arguments passed on non-zero stages, iirc.  PipelineStage would have an if/else for getting args from user vs buffers.

We should probably also add validation there that if users pass an arg to non-zero stage we complain that it will be ignored!

Reply:
yeah, that if-else is a hard wire, which we could relax if needed.